### ISSUE TITLE
[feature] Support semver style versioning

### ISSUE THREAD

**Issue by asraa** [sigstore/cosign-installer#111]
**Description**

Thank you for this! We are using this project inside our workflow, and were using the project with no flags, which fetched our version at main. Due to the major version bump, our workflows broke because of the incompatible flags.

Would it be possible to support something like v2.0 that would fetch the latest patch version? This way we can recieve updated patches, but ensure not to break on any major version bump.

cc @laurentsimon @ianlewis

**Comment by cpanato** [comment]
we released v3 that is using cosign v2.
the v2 version of cosign-installer defaults to cosign v1.13.1.

please, as a good practice never use main, only if you control the repo or like to live yolo :D

**Comment by asraa** [comment]
yes! we just realized that we didn't break our users - only our tests @main broke :)

so i think this is all good in terms of being stable.

let's close this - i think doing a semver style would be annoying, since IF there are critical patches, let's have dependants update and release, instead of possibly updating the version under the hood.
